### ISSUE TITLE
rust-lld: error: unable to find library -lstdc++

### ISSUE THREAD

**Issue by proohit** [rustwasm/wasm-bindgen#2536]
### Summary
First of all, I do not know, if that's the right place to ask this question, as I'm also very new to Rust/WASM.

My goal is to build a wasm module which exposes a simple function to js which fits a model and calculates and returns it's accuracy.

I copied the [rustlearn example](https://github.com/maciejkula/rustlearn#logistic-regression) into the [wasm-pack rust to js tutorial project](https://rustwasm.github.io/book/game-of-life/introduction.html). There, I published the example as a function

```rust
mod utils;

use wasm_bindgen::prelude::*;

// When the `wee_alloc` feature is enabled, use `wee_alloc` as the global
// allocator.
#[cfg(feature = "wee_alloc")]
#[global_allocator]
static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;

#[wasm_bindgen]
extern {
    #[wasm_bindgen(js_namespace = console)]
    fn log(s: &str);
}

#[wasm_bindgen]
pub fn greet() {
    log("Hello, nodejs-rust!");
}

use rustlearn::prelude::*;
use rustlearn::datasets::iris;
use rustlearn::cross_validation::CrossValidation;
use rustlearn::linear_models::sgdclassifier::Hyperparameters;
use rustlearn::metrics::accuracy_score;

#[wasm_bindgen]
pub fn fit_accuracy()-> f32 {
    let (x, y) = iris::load_data();

    let num_splits = 10;
    let num_epochs = 5;
    
    let mut accuracy = 0.0;
    
    for (train_idx, test_idx) in CrossValidation::new(x.rows(), num_splits) {
    
        let x_train = x.get_rows(&train_idx);
        let y_train = y.get_rows(&train_idx);
        let x_test = x.get_rows(&test_idx);
        let y_test = y.get_rows(&test_idx);
    
        let mut model = Hyperparameters::new(x.cols())
                                        .learning_rate(0.5)
                                        .l2_penalty(0.0)
                                        .l1_penalty(0.0)
                                        .one_vs_rest();
    
        for _ in 0..num_epochs {
            model.fit(&x_train, &y_train).unwrap();
        }
    
        let prediction = model.predict(&x_test).unwrap();
        accuracy += accuracy_score(&y_test, &prediction);
    }
    
    accuracy /= num_splits as f32;
    return accuracy;
}
```
When building with wasm-pack to target nodejs, I get the following error:

```
direnc@direnc-VirtualBox:~/workspace/nodejs-rust$ wasm-pack build --target nodejs
[INFO]: Checking for the Wasm target...
[INFO]: Compiling to Wasm...
   Compiling nodejs-rust v0.1.0 (/home/direnc/workspace/nodejs-rust)
warning: function is never used: `set_panic_hook`
 --> src/utils.rs:1:8
  |
1 | pub fn set_panic_hook() {
  |        ^^^^^^^^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

error: linking with `rust-lld` failed: exit code: 1
  |
  = note: "rust-lld" "-flavor" "wasm" "--rsp-quoting=posix" "-z" "stack-size=1048576" "--stack-first" "--allow-undefined" "--fatal-warnings" "--no-demangle" "--no-entry" "--export-dynamic" "-L" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib" "-L" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/self-contained" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.0.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.1.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.2.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.3.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.4.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.5.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.6.rcgu.o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.nodejs_rust.7l8m517g-cgu.7.rcgu.o" "-o" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.wasm" "--export" "__wbindgen_describe___wbg_log_3facba13162b5bc0" "--export" "__wbindgen_describe_fit_accuracy" "--export" "__wbindgen_describe_greet" "--export" "fit_accuracy" "--export" "greet" "--export" "__externref_drop_slice" "--export" "__externref_heap_live_count" "--export" "__externref_table_alloc" "--export" "__externref_table_dealloc" "--export" "__wbindgen_exn_store" "--export" "__wbindgen_free" "--export" "__wbindgen_malloc" "--export" "__wbindgen_realloc" "--export" "__wbindgen_describe___wbg_error_4bb6c2a97407129a" "--export" "__wbindgen_describe___wbg_new_59cb74e423758ede" "--export" "__wbindgen_describe___wbg_stack_558ba5917b466edd" "--export=__heap_base" "--export=__data_end" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/nodejs_rust.521rxx7eek9bx6p9.rcgu.o" "--gc-sections" "--no-entry" "-O2" "-L" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps" "-L" "/home/direnc/workspace/nodejs-rust/target/release/deps" "-L" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/build/rustlearn-3659aa6d1c03cfb5/out" "-L" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/libconsole_error_panic_hook-d948fd9f6890c123.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/libcfg_if-a2cf87716cf74ebd.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/librustlearn-b5b0c32c27a75715.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/libserde-c5c6dbad6b012958.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/librand-7c3715fb489da840.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/librand-34fe4ef2d35d1499.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/libcrossbeam-d303f01b0717aeee.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/libwasm_bindgen-b16cfe1f6cc3840d.rlib" "/home/direnc/workspace/nodejs-rust/target/wasm32-unknown-unknown/release/deps/libcfg_if-5b2b6458a8a1968f.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libstd-5794ce00664e7d4f.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libpanic_abort-0af2cb904b1c0ce4.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libdlmalloc-089cbb1e0858411f.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/librustc_demangle-59b9177011348216.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libhashbrown-004608aa63d8173f.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/librustc_std_workspace_alloc-0360b3804bee2a5c.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libunwind-84c40d9b356616dc.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libcfg_if-f4c142fbb6be3928.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/liblibc-a03bd8d7d65362c3.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/liballoc-0ed0dde53ea46b32.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/librustc_std_workspace_core-b85906c5e92e0d09.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libcore-cfab44fd0fde436d.rlib" "/home/direnc/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libcompiler_builtins-7716ccb56cc0a574.rlib" "-l" "stdc++"
  = note: rust-lld: error: unable to find library -lstdc++
          

error: aborting due to previous error; 1 warning emitted

error: could not compile `nodejs-rust`

To learn more, run the command again with --verbose.
Error: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit code: 101
  full command: "cargo" "build" "--lib" "--release" "--target" "wasm32-unknown-unknown"
```

When running from a `main.rs` file with `cargo run`, the function works correctly (putting aside all the js bindings of course).

### Additional Details

Running on VirtualBox, Ubuntu 20.04
```
direnc@direnc-VirtualBox:~/workspace/nodejs-rust$ cargo --version
cargo 1.51.0 (43b129a20 2021-03-16)
```
```
direnc@direnc-VirtualBox:~/workspace/nodejs-rust$ rustc --version
rustc 1.51.0 (2fd73fabe 2021-03-23)
```

**Comment by alexcrichton** [comment]
Thanks for the report! I don't think this is a bug with wasm-bindgen though. This is either a dependency pulling in `-lstdc++` or something in your environment configuring that (e.g. via `RUSTFLAGS`).
